### PR TITLE
Ntf 39 docker-compose

### PR DESCRIPTION
**Pull request by qjflores**
uses docker-compose to connect postgres database

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    build: .
+    image: bluefin-link
+    environment:
+      - AWS_ACCESS_KEY_ID=$AWS_ACCESS_KEY_ID
+      - AWS_DEFAULT_OUTPUT=$AWS_DEFAULT_OUTPUT
+      - AWS_DEFAULT_REGION=$AWS_DEFAULT_REGION
+      - AWS_SECRET_ACCESS_KEY=$AWS_SECRET_ACCESS_KEY
+      - CODECOV_TOKEN=$CODECOV_TOKEN
+      - NPM_AUTH_TOKEN=$NPM_AUTH_TOKEN
+      - DATABASE_URL=postgres://postgres:posgres@pg:5432/test
```
Comment (qjflores) [inline docker-compose.yml:13]: this is required or else the bluefin-link service won't connect

File: docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    depends_on:
```
Comment (qjflores) [inline docker-compose.yml:14]: this sets up the networking for us

File: package.json [hunk 3]
```diff
@@ -29,5 +29,16 @@
     "nyc": "~13.1.0",
     "sinon": "~7.2.3",
     "standard": "~12.0.1"
+  },
+  "ava": {
+    "files": [
+      "test/*.js",
+      "!test/lib/**/*.js",
```
Comment (qjflores) [inline package.json:36]: excludes the files in the test folder that are helpers

File: package.json [hunk 4]
```diff
@@ -29,5 +29,16 @@
...
+      "!test/manual/**/*.js"
```
Comment (qjflores) [inline package.json:37]: excludes the files in the test folder that are helpers

File: docker-compose.yml [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+      - pg
+
+  pg:
+    image: postgres:10-alpine
+    ports:
+      - 5432:5432
+    environment:
+      - POSTGRES_USER=postgres
```
Comment (crankycookie) [inline docker-compose.yml:22]: So I think we can leave the other connection strings in the test as `postgres:///test` if we just add these parameters to the environment:
```
- PGHOST=pg
- PGPORT=5432
- PGUSER=postgres
- PGPASS=postgres
```

**Review by crankycookie — DISMISSED** [review]
Just one requested change so we don't have to change all the test files

File: docker-compose.yml (hunk 5 above)
Comment (qjflores) [reply docker-compose.yml:22]: That’s seems possible.

The changes to the connection string follow the same format that we’d use to connect to aws. 

There are currently no unit tests for creating and validating a connection string, I've created a ticket for it. It'd be nice to have but outside of the scope of the issue we're addressing.

https://themuse.atlassian.net/browse/NTF-101

File: docker-compose.yml [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+      - AWS_DEFAULT_REGION=$AWS_DEFAULT_REGION
+      - AWS_SECRET_ACCESS_KEY=$AWS_SECRET_ACCESS_KEY
+      - CODECOV_TOKEN=$CODECOV_TOKEN
+      - NPM_AUTH_TOKEN=$NPM_AUTH_TOKEN
+      - DATABASE_URL=postgres://postgres:posgres@pg:5432/test
+    depends_on:
+      - pg
+
+  pg:
+    image: postgres:10-alpine
```
Comment (davidkaminsky) [inline docker-compose.yml]: Should we upgrade to 11?

Comment (qjflores) [reply docker-compose.yml]: upgrading to 11 works

**Comment by davidkaminsky** [comment]
It still seems to think the tests are failing, fyi.

**Comment by qjflores** [comment]
@davidkaminsky the tests were failing because of a different change.

it's fixed with https://github.com/dailymuse/bluefin-link/pull/15/commits/ecb240c5cba72e02f90b6bc84dbf5dd4218c790b

**Review by crankycookie — APPROVED** [review]

**Pull request merged**